validation-test/stdlib/Data.swift

```diff
@@ -35,11 +35,33 @@ DataTestSuite.test("associated types") {
   expectRandomAccessCollectionAssociatedTypes(
     collectionType: Subject.self,
     iteratorType: Data.Iterator.self,
-    subSequenceType: MutableRangeReplaceableRandomAccessSlice<Subject>.self,
+    subSequenceType: Subject.self,
     indexType: Int.self,
     indexDistanceType: Int.self,
     indicesType: CountableRange<Int>.self)
 }
 
+DataTestSuite.test("Data SubSequence") {
```

Comment:
checkRandomAccessCollection creates slices of slices iirc, I would prefer to add more tests in the unit tests and not the validation; did you have a specific test in mind?
e.g. something to do with something like this? `Data(bytes: [1, 2, 3, 4, 5, 6, 7, 8, 9])[0..<9][1..<4]`